### DISCUSSION TITLE
SPF module

### DISCUSSION THREAD

**Discussion by jsaathof** [rspamd/rspamd#4819] [Q&A]
HI,

I'm a new (and very happy) user converted from spamassassin. I really like the simplicity and the GUI of rspamd, and I have it running for only a couple of hours now. 

I did have an error when starting the configwizard:
```
init of /usr/share/rspamd/lualib/lua_ffi/spf.lua failed: /usr/share/rspamd/lualib/lua_ffi/spf.lua:71: size of C type is unknown or too large at line 35; trace: [1]:{[C]:-1 - cdef [C]}; [2]:{/usr/share/rspamd/lualib/lua_ffi/spf.lua:71 - <unknown> [main]};
```

The list of modules reports the SPF module indeed as disabled:
```
Modules disabled (failed): spf
```
I'm running openSUSE Leap 15.5 and installed the package that comes with it, which is version 3.8.1. Am I doing something wrong or am I missing something? I have no experience with the Lua language. The version of Lua that was installed with the package is 5.3, although the libraries (mainly librspamd-server.so) seem to have been built against 5.1.

I hope anyone can shed some light on this issue and help me out.

**Comment by fatalbanana** [comment]
That's a bug & it's particular to the build you're using; consider using [Rspamd in Docker](https://hub.docker.com/r/rspamd/rspamd), or on a distribution targeted by official packages.

[The apparent reason is that PLUGINSDIR is incorrectly set at build time and so Rspamd tries to execute things it shouldn't as plugins](https://github.com/rspamd/rspamd/issues/4977#issuecomment-2227294583)

[The solution could be to avoid setting it entirely](https://github.com/rspamd/rspamd/discussions/4819#discussioncomment-10417508)

**Comment by jsaathof** [comment]
I was afraid that was going to be the answer. What is the cause of this? Is it indeed the Lua version against which it is compiled?

**Reply by M-Stenzel** [reply]
Hi, did you manage to solve the issue?
I have the very same problem,

Martin.

**Comment by jsaathof** [comment]
No I haven't sadly. It is still present on Leap 15.6. But I don't really understand what the exact issue is. The issue is said to be the PLUGINDIR setting which is indeed set in this build from openSUSE. But it's not that the plugin is not in the PLUGINDIR. There seems to be something else wrong.

**Comment by jsaathof** [comment]
I have now recompiled rspamd myself on Leap 15.6 because it did not include the latest version (only 3.8.4 is in the repo). I have not changed the plugin path and now the SPF module does not give me an error.
